tests/vde_backend/test_rtu.py

```diff
@@ -0,0 +1,120 @@
+                resp.status_code = 200
+                resp.reason = 'OK'
+                resp._content = mock_rtu_pages.DATA_PAGE.encode(resp.encoding)
+        else:
+            resp.status_code = 404
+            resp.reason = 'Not Found'
+    else:
+        resp.status_code = 401
+        resp.reason = 'Unauthorized'
+
+    return resp
+
+class TestRTUDriver(unittest.TestCase):
+    @unittest.mock.patch('requests.Session.get', side_effect=mocked_requests_get)
+    def test_rtu_init_wrong_creds(self, mock_get):
+        self.assertRaises(
+            requests.HTTPError,
+            rtu_comm.RTUConnector,
+            '192.168.0.1',
+            'test',
+            'blah',
+            n_retries=2
+        )
+
+    @unittest.mock.patch('requests.Session.get', side_effect=mocked_requests_get)
+    def test_rtu_init_right_creds(self, mock_get):
+        rtu = rtu_comm.RTUConnector('192.168.0.1', 'Test1', 'Test2')
+        self.assertEqual(
+            'http://192.168.0.1/ABBRTU560/hwTree_pdInfoMon?IDNR=0&REF=549&MODE=1',
+            rtu.url_hwinfo,
+        )
+
+    @unittest.mock.patch('requests.Session.get', side_effect=mocked_requests_get)
+    def test_rtu_read_values_tiv(self, mock_get):
+        rtu = rtu_comm.RTUConnector('192.168.0.1', 'Test1', 'Test2')
+        test_vals = rtu.read_values().to_dict()
+        self.assertEqual('CET', test_vals['ts'].tz.zone)
+        num_vals = { k: v for k, v in test_vals.items() if k != 'ts' }
+        self.assertDictEqual(
+            {
+                'COS PHI': -0.906700,
+                'PUISSANCE ACTIVE': 30720.000000,
+                'PUISSANCE APPARENTE': 34920.000000,
+                'PUISSANCE REACTIVE': -6120.000000,
+                'TENSION PHASE 1-2': 237.500000,
+                'TENSION PHASE 2-3': 237.229996,
+                'TENSION PHASE 3-1': 237.319992,
+            },
+            num_vals,
+        )
+
+    def test_prepare_rtu_row(self):
+        test_vals = pd.Series({
+            'COS PHI': -0.906700,
+            'PUISSANCE ACTIVE': 30720.000000,
+            'PUISSANCE APPARENTE': 34920.000000,
+            'PUISSANCE REACTIVE': -6120.000000,
+            'TENSION PHASE 1-2': 237.500000,
+            'TENSION PHASE 2-3': 237.229996,
+            'TENSION PHASE 3-1': 237.319992,
+            'ts': pd.Timestamp('2022-09-11T17:27:00', tz='CET'),
+        })
+        prep_vals = sync_rtu.prepare_rtu_row(test_vals, '1.1.1.1')
+        self.assertDictEqual(
+            {
+                'ip': '1.1.1.1',
+                'day': '2022-09-11',
+                'ts': pd.Timestamp('2022-09-11T17:27:00', tz='CET'),
+                'cos_phi': -0.906700,
+                'active': 30720.000000,
+                'apparent': 34920.000000,
+                'reactive': -6120.000000,
+                'tension1-2': 237.500000,
+                'tension2-3': 237.229996,
+                'tension3-1': 237.319992,
+            },
+            prep_vals.to_dict()
+        )
```

Comment:
Is it done on purpose to have 2 different sets of index names ? Because we could have converted the index names directly in read_values().

Reply:
The conversion needs to happen at some point because the RTU is giving different columns than what we need.
I made the choice of having the RTU driver return raw values, as given by the machine and letting the responsibility of the conversion to the script that uses it. This way, the table, the column names and the conversion all happen in the same file. It is easier to understand, debug, and if a change must happen. For instance, if you want to rename the `active` column into `active_power`, there is only one file to change, vs. 2 if the RTU was doing the conversion.